src/Aquifer.API/Endpoints/Resources/Content/NeedsTranslation/List/Endpoint.cs

```diff
@@ -0,0 +1,139 @@
+        {
+            ResourceContents = resourceContent,
+            Total = total
+        };
+
+        await SendOkAsync(response, ct);
+    }
+
+    private static string BuildQuery(Request req, UserEntity user, bool getTotalCount, bool hasSearchQuery, bool isExactSearch)
+    {
+        const string selectCount = "SELECT COUNT(DISTINCT(RC.Id)) AS Count";
+        const string selectProperties = """
+                                        SELECT RC.Id AS Id, R.EnglishLabel, PR.DisplayName AS ParentResourceName,
+                                        COALESCE(RCV.SourceWordCount, RCV.WordCount) AS WordCount
+                                        """;
+
+        var groupingWithOffset = $"""
+                                  GROUP BY RC.Id, R.EnglishLabel, PR.DisplayName, L.EnglishDisplay, COALESCE(RCV.SourceWordCount, RCV.WordCount), R.SortOrder
+                                  ORDER BY PR.DisplayName, R.SortOrder, R.EnglishLabel
+                                  OFFSET {req.Offset} ROWS FETCH NEXT {req.Limit} ROWS ONLY
+                                  """;
+
+        return $"""
+                {(getTotalCount ? selectCount : selectProperties)}
+                FROM ResourceContents RC
+                    INNER JOIN Resources R ON R.Id = RC.ResourceId
+                    INNER JOIN ParentResources PR ON PR.id = R.ParentResourceId
+                    INNER JOIN Languages L ON L.Id = RC.LanguageId
+                    INNER JOIN ResourceContentVersions RCV ON RCV.ResourceContentId = RC.Id
+                WHERE RC.MediaType != {(int)ResourceContentMediaType.Audio}
+                {ApplyLanguageIdFilter(user.LanguageId ?? 1)}
+                {ApplyParentResourceIdFilter(req.ParentResourceId)}
+                {ApplySearchQueryFilter(hasSearchQuery, isExactSearch)}
+                {ApplyBookAndChapterFilter(req.BookCode, req.StartChapter, req.EndChapter)}
+                {(getTotalCount ? "" : groupingWithOffset)}
+                """;
+    }
+
+    private static string ApplyLanguageIdFilter(int languageId)
+    {
+        return $"""
+                AND RC.LanguageId = 1 AND RCV.IsPublished = 1 AND NOT EXISTS (
+                    SELECT 1 FROM ResourceContents RCL
+                    WHERE RCL.ResourceId = R.Id AND RCL.LanguageId = {languageId}
+                )
+                """;
+    }
+
+    private static string ApplyParentResourceIdFilter(int? parentResourceId)
+    {
+        return parentResourceId.HasValue ? $"AND R.ParentResourceId = {parentResourceId.Value}" : "";
+    }
+
+    private static string ApplyBookAndChapterFilter(string? bookCode, int? startChapter, int? endChapter)
+    {
+        var verseRange = BibleUtilities.VerseRangeForBookAndChapters(bookCode, startChapter, endChapter);
+        if (verseRange is null)
+        {
+            return "";
+        }
+
+        var startVerseId = verseRange.Value.Item1;
+        var endVerseId = verseRange.Value.Item2;
```

Comment:
I suggest updating `BibleUtilities.VerseRangeForBookAndChapters()` [here](BibleUtilities.VerseRangeForBookAndChapters) to return a `ValueTuple` with named items (like the other methods in that class [such as this one](https://github.com/BiblioNexusStudio/aquifer-server/blob/bf045f44c56bd5ce8b272da67a218b3dbe85f792/src/Aquifer.Common/Utilities/BibleUtilities.cs#L77)) so that you can refer to them by name here instead of `Item1`/`Item2`.